django/db/backends/mysql/schema.py

```diff
@@ -32,8 +32,10 @@ def quote_value(self, value):
     def skip_default(self, field):
         """
         MySQL doesn't accept default values for some data types and implicitly
-        treats these columns as nullable.
+        treats these columns as nullable. MariaDB 10.2+ fixes this.
         """
+        if self.connection.mysql_is_mariadb and self.connection.mysql_version[:2] > (10, 2):
+            return False
```

Comment:
Shouldn't this be a feature flag and to allow a Maria DB specific backend to override it?